### DISCUSSION TITLE
Azure AD - Error during authentication

### DISCUSSION THREAD

**Discussion by kaczmareck-quentin** [outline/outline#6789] [Self Hosting]
Impossible to login using Entra ID.

In Entra the connection is shown as successful
Set cookie for /azure is showing the right domain

state=outline.mydomain.example|03XXXXXXXX|web; path=/; expires=Thu, 11 Apr 2024 07:27:29 GMT; domain=outline.mydomain.example; secure; httponly

Error :
`{"error":"Cannot read properties of undefined (reading '0')","level":"error","message":"Error during authentication","stack":"TypeError: Cannot read properties of undefined (reading '0')\n    at Strategy._verify (/opt/outline/build/plugins/azure/server/auth/azure.js:50:54)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}
{"label":"http","level":"info","message":"  --> GET /auth/azure.callback?code=0.XXXXXXXX`

https://graph.microsoft.com/v1.0/me
![https://graph.microsoft.com/v1.0/me](https://github.com/outline/outline/assets/148951309/ce4a5df1-f729-4f15-bcbf-4ba1736b7087)

https://graph.microsoft.com/v1.0/organization
![https://graph.microsoft.com/v1.0/organization](https://github.com/outline/outline/assets/148951309/0d86a9ad-5bfe-4f55-93ea-bb2f81a070f8)

Also inserted into the "teams" and "authentication_providers" using an oidc install as a base, but still the same error.

**Comment by tommoor** [comment]
Weird, you can see the related code here

https://github.com/outline/outline/blob/main/plugins/azure/server/auth/azure.ts#L76

It's over three years old and working for thousands of installations 🤔 . Only thing I can think of is there is some sort of proxy between Outline server and Azure that's returning a response, but not the expected JSON.

**Reply by kaczmareck-quentin** [reply]
using this as a nginx conf : 
```
server {
    listen 80;
    listen [::]:80;

    server_name outline.domain.example;

    return 301 https://$host$request_uri;
}
server {
  server_name outline.domain.example;
  listen 80;
  listen 443 ssl;

  ssl_certificate /certs/public.cer;
  ssl_certificate_key /certs/private.key;
  keepalive_timeout 70;
  ssl_protocols TLSv1.3 TLSv1.2;

  # SSL certificate configuration is not included in this example

  location / {
    proxy_pass http://localhost:3000/;

    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Host $host;

    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;proxy_set_header Host $host;
    proxy_set_header Host $http_host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Scheme $scheme;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_redirect off;
  }
}

```
